source/lac/trilinos_sparse_matrix.cc

```diff
@@ -2651,6 +2726,114 @@ namespace TrilinosWrappers
 
 
 
+      TrilinosPayload::TrilinosPayload(
+        const TrilinosPayload &                   payload_exemplar,
+        const TrilinosWrappers::PreconditionBase &preconditioner)
+        : use_transpose(payload_exemplar.UseTranspose())
+        , communicator(payload_exemplar.get_mpi_communicator())
+        , domain_map(
+            payload_exemplar.locally_owned_domain_indices().make_trilinos_map(
+              communicator.Comm()))
+        , range_map(
+            payload_exemplar.locally_owned_range_indices().make_trilinos_map(
+              communicator.Comm()))
```

Comment:
It seems that the lambdas in this function are pretty similar to the ones in the `Payload` constructed from the two `PreconditionerBase` objects in the previous function. Do you see a possibility to make both use the same code (with some adaptations, so we avoid adding around 100 new lines of code with 90% as in the other function. (I realize I do not see an immediate solution, so feel free to ignore it.)

Reply:
@kronbichler I've still got to run some tests locally to make sure that I didn't make any errors, but I'm sure that you'll appreciate the simplification of the constructors that this solution aims to achieve.